src/test/java/net/snowflake/client/SfTest.java

```diff
@@ -0,0 +1,12 @@
+package net.snowflake.client;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.TYPE})
+public @interface SfTest {
```

Comment:
annotation could be put on class, not only method - is this class aware of it? 

are we able to handle different version on both levels? e.g. class declares availability since 3.14.5 whereas method declares availability for version 3.14.6

another question is that during implemntation we are not sure which version will be released so the given version should be last version that **does not** have the tested feature